src/legacy/server/logging/rotate/log_rotator.ts

```diff
@@ -150,12 +152,20 @@ export class LogRotator {
   }
 
   async _startLogFileSizeMonitor() {
-    this.usePolling = await this._shouldUsePolling();
+    this.usePolling = this.config.get('logging.rotate.usePolling');
+    this.shouldUsePolling = await this._shouldUsePolling();
 
-    if (this.usePolling && this.usePolling !== this.config.get('logging.rotate.usePolling')) {
+    if (this.usePolling && !this.shouldUsePolling) {
       this.log(
         ['warning', 'logging:rotate'],
-        'The current environment does not support `fs.watch`. Falling back to polling using `fs.watchFile`'
+        'Looks like your current environment support `fs.watch` but you are using polling which uses `fs.watchFile`'
```

Comment:
I think users are not aware of fs specificity. We can say that their OS supports a faster algorithm than slow polling.

Reply:
I agree :)